exercises/react/src/react.h

```diff
@@ -0,0 +1,29 @@
+#ifndef REACT_H
+#define REACT_H
+
+struct reactor;
+struct cell;
+
+typedef int (*compute1) (int);
+typedef int (*compute2) (int, int);
+
+struct reactor *create_reactor();
+// free_reactor should free all cells created under that reactor.
+void free_reactor(struct reactor *);
+
+struct cell *create_input_cell(struct reactor *, int initial_value);
+struct cell *create_compute1_cell(struct reactor *, struct cell *, compute1);
+struct cell *create_compute2_cell(struct reactor *, struct cell *,
+                                  struct cell *, compute2);
+
+int cell_value(struct cell *);
+void cell_value_set(struct cell *, int new_value);
+
+typedef void (*callback) (void *, int);
+typedef int callback_id;
+
+// The callback should be called with the same void * given in add_callback.
```

Comment:
note: my way of dealing with no closure capture. if I didn't do this, I'd have to have at least three functions that do the same thing but on three different global variables, since one test requires having three callbacks.